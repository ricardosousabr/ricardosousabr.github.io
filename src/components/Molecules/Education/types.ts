export interface IEducationProps {
  title: string;
  subtitle: string;
  faculty: string;
  course: string;
  state: string;
}