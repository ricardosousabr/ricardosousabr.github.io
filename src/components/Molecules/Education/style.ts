import styled from "styled-components";

export const EducationStyle = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-between;
  background: #030712;
  padding: 50px 0;

`
export const BoxText = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-between;
  padding: 30px 0;
`
export const BoxTitle = styled.div`
  padding-bottom: 20px;
`

export const BoxTextInformation = styled.div`
  display: flex;
  align-items: left;
  flex-direction: column;
  justify-content: space-between;
  height: 100px;

`
