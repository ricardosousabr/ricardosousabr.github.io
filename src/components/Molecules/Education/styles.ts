import styled from "styled-components";

export const Wrapper = styled.div`
  background: #282C33;
  padding: 50px 0;

`

export const EducationStyle = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-between;


`
export const BoxText = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-between;
  padding: 30px 0;
  text-align: center;
`
export const BoxTitle = styled.div`
  padding-bottom: 20px;
`

export const BoxTextInformation = styled.div`
  display: flex;
  align-items: left;
  flex-direction: column;
  justify-content: space-between;
  height: 100px;

`
