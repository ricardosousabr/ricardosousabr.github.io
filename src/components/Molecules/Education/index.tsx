import Text from "../../Atoms/Text";
import Container from "../../Atoms/Container";
import { IEducationProps } from "./types";
import { EducationStyle, Wrapper, BoxText, BoxTitle, BoxTextInformation } from "./styles";

export default function Education({ title, subtitle, faculty, course, state }:IEducationProps) {
  return (
    <Wrapper>
      <Container>
        <EducationStyle>
          <BoxText>
            <BoxTitle>
              <Text color="white" fontSize="20px">{ title }</Text>
            </BoxTitle>
            <div>
              <Text color="white" fontSize="16px">{ subtitle }</Text>
            </div>
          </BoxText>
          <BoxTextInformation>
            <Text color="#66BEF5" fontSize="16px">{ faculty }</Text>
            <Text color="white" fontSize="16px">{ course }</Text>
            <Text color="white" fontSize="16px">{ state }</Text>
          </BoxTextInformation>
        </EducationStyle>
      </Container>
    </Wrapper>
  )
}
