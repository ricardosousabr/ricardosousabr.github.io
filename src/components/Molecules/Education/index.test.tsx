import renderer from "react-test-renderer";
import Education from "./index";
import locale from "../../../locales";

test("Link renders correctly", () => {
  const { education } = locale['pt-br'];
  const tree = renderer.create(<Education {...education} />).toJSON();
  expect(tree).toMatchSnapshot();
});
