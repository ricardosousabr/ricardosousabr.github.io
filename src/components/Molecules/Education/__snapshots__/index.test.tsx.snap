// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div
  className="sc-gtsqUy dZrVMS"
>
  <div
    className="sc-dlniIP kchgDo"
  >
    <div
      className="sc-hKFymg gLhFjD"
    >
      <p
        className="sc-bdnyFh iOsQJx"
        color="white"
        fontSize="20px"
      >
        Education
      </p>
    </div>
    <div>
      <p
        className="sc-bdnyFh iOusbc"
        color="white"
        fontSize="16px"
      >
        These are my qualifications
      </p>
    </div>
  </div>
  <div
    className="sc-eCAqax dLndRX"
  >
    <p
      className="sc-bdnyFh jKeQRk"
      color="#66BEF5"
      fontSize="16px"
    >
      Uninassau
    </p>
    <p
      className="sc-bdnyFh iOusbc"
      color="white"
      fontSize="16px"
    >
      Analise e desenvolvimento de sistemas
    </p>
    <p
      className="sc-bdnyFh iOusbc"
      color="white"
      fontSize="16px"
    >
      State: Cursando
    </p>
  </div>
</div>
`;
